components/amount-buttons.jsx

```diff
@@ -142,8 +140,8 @@ var AmountButtons = React.createClass({
     var amount = this.props.amount;
     var presets = this.props.presets;
     var preset = presets.indexOf(amount);
-    var userInputting = this.state.userInputting;
-    if (userInputting) {
+    var otherChecked = this.state.userInputting || (amount && preset < 0);
```

Comment:
This is all we need to add to get the same results as the code removed in getInitialState.